sway-core/src/error.rs

```diff
@@ -876,8 +877,12 @@ pub enum CompileError {
         missing_patterns: String,
         span: Span,
     },
-    #[error("Impure function called inside of pure function. Pure functions can only call other pure functions. Try making the surrounding function impure by prepending \"impure\" to the function declaration.")]
-    PureCalledImpure { span: Span },
+    #[error(
+        "Impure function called inside of pure function. Pure functions can only call \
+        other pure functions. Try making the surrounding function impure by adding \
+        \"#[{STORAGE_PURITY_ATTRIBUTE_NAME}({attrs})]\" to the function declaration."
```

Comment:
Ideally this error message would more precisely reflect the use annotation usage rather than referring to "impure" and "pure."